app/Http/Controllers/PostController.php

```diff
@@ -1,18 +1,49 @@
 <?php
 
 namespace App\Http\Controllers;
+use App\Location;
+use App\Post;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Storage;
 
 class PostController extends Controller
 {
 
-    public function create()
+    public function create($id)
     {
-        return view('posts.create');
+        //$location = Location::with('posts')->findOrFail($id);
+
+        return view('posts.create')->with("location", $id);
     }
 
-    public function store()
+    public function store(Request $request, $id)
     {
 
-    }
+        $location = Location::findOrFail($id);
+
+        //validate data ~ TODO: move to configuration file
+        $request->validate([
+            'body' => 'required|max:50000',
+            'image' => 'max: 2048'
+        ]);
+
+        //create post
+        $post = new Post();
+        $post->body = $request['body'];
+
+        $post->author_id = 1;
+        $post->author_type = "App\User";
```

Comment:
Whenever you're doing class accesses use App\User::class, rather than a string. Allows things to be refactored easier because you're working with tokens, not random strings everywhere.